### ISSUE TITLE
Breaks with compose 1.0.0-beta05

### ISSUE THREAD

**Issue by matejdro** [Zhuinden/simple-stack-compose-integration#6]
Attempting to start compose app with your plugin and compose 1.0.0-beta05 crashes with the following exception:

```
    java.lang.NoSuchFieldError: No static field Companion of type Landroidx/compose/foundation/layout/BoxScope$Companion; in class Landroidx/compose/foundation/layout/BoxScope; or its superclasses (declaration of 'androidx.compose.foundation.layout.BoxScope' appears in /data/app/~~zFpvp152yTYcEyZDvW-v6A==/com.matejdro.app-alvOcdgEeyD-xUC5GdNEoQ==/base.apk)
        at com.zhuinden.simplestackcomposeintegration.core.AnimatingComposeStateChanger$BackstackState$RenderScreen$1$1$1.invoke(ComposeIntegrationCore.kt:384)
        at com.zhuinden.simplestackcomposeintegration.core.AnimatingComposeStateChanger$BackstackState$RenderScreen$1$1$1.invoke(ComposeIntegrationCore.kt:230)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
        at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:193)
        at androidx.compose.runtime.saveable.SaveableStateHolderImpl.SaveableStateProvider(SaveableStateHolder.kt:84)
        at com.zhuinden.simplestackcomposeintegration.core.AnimatingComposeStateChanger$BackstackState.RenderScreen(ComposeIntegrationCore.kt:230)
        at com.zhuinden.simplestackcomposeintegration.core.AnimatingComposeStateChanger.RenderScreen(ComposeIntegrationCore.kt:275)
```

**Comment by Zhuinden** [comment]
Huh, it appears I'll need to update this to beta05 so that there would be no version mismatch against compose. 🤔 I'll try to do it asap

**Comment by Zhuinden** [comment]
I updated to beta06 and then got this

```
2021-05-06 05:26:44.014 3530-3530/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.zhuinden.simplestackcomposesimpleexample, PID: 3530
    java.lang.IllegalArgumentException: Invalid start index
        at androidx.compose.runtime.Pending.<init>(Composer.kt:91)
        at androidx.compose.runtime.ComposerImpl.start(Composer.kt:1821)
        at androidx.compose.runtime.ComposerImpl.startReplaceableGroup(Composer.kt:1000)
        at com.zhuinden.simplestackcomposeintegration.core.SimpleComposeStateChanger$BackstackState.RenderScreen(ComposeIntegrationCore.kt:312)
        at com.zhuinden.simplestackcomposeintegration.core.SimpleComposeStateChanger.RenderScreen(ComposeIntegrationCore.kt:332)
        at com.zhuinden.simplestackcomposeintegration.core.SimpleComposeStateChanger$RenderScreen$1.invoke(Unknown Source:10)
        at com.zhuinden.simplestackcomposeintegration.core.SimpleComposeStateChanger$RenderScreen$1.invoke(Unknown Source:10)
        at androidx.compose.runtime.RecomposeScopeImpl.compose(RecomposeScopeImpl.kt:96)
        at androidx.compose.runtime.ComposerImpl.recomposeToGroupEnd(Composer.kt:2116)
        at androidx.compose.runtime.ComposerImpl.skipCurrentGroup(Composer.kt:2374)
        at androidx.compose.runtime.ComposerImpl.doCompose(Composer.kt:2516)
        at androidx.compose.runtime.ComposerImpl.recompose$runtime_release(Composer.kt:2487)
        at androidx.compose.runtime.CompositionImpl.recompose(Composition.kt:546)
        at androidx.compose.runtime.Recomposer.performRecompose(Recomposer.kt:733)
        at androidx.compose.runtime.Recomposer.access$performRecompose(Recomposer.kt:102)
        at androidx.compose.runtime.Recomposer$runRecomposeAndApplyChanges$2$2.invoke(Recomposer.kt:443)
        at androidx.compose.runtime.Recomposer$runRecomposeAndApplyChanges$2$2.invoke(Recomposer.kt:415)
        at androidx.compose.ui.platform.AndroidUiFrameClock$withFrameNanos$2$callback$1.doFrame(AndroidUiFrameClock.android.kt:34)
        at androidx.compose.ui.platform.AndroidUiDispatcher.performFrameDispatch(AndroidUiDispatcher.android.kt:109)
        at androidx.compose.ui.platform.AndroidUiDispatcher.access$performFrameDispatch(AndroidUiDispatcher.android.kt:41)
        at androidx.compose.ui.platform.AndroidUiDispatcher$dispatchCallback$1.doFrame(AndroidUiDispatcher.android.kt:69)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:964)
        at android.view.Choreographer.doCallbacks(Choreographer.java:790)
        at android.view.Choreographer.doFrame(Choreographer.java:721)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:951)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```

I'm honestly not sure why, and I might end up removing `SimpleComposeStateChanger` in favor of the animating one.

Who wants screen changes without animations anyway? 🤔

**Comment by Zhuinden** [comment]
Should be fixed with 0.3.0?

Please note I had to make an API breaking change for stability sake.

https://github.com/Zhuinden/simple-stack-compose-integration/commit/7670a13d3f6f64a0dd412a1c183421737a08b8a6

**Comment by matejdro** [comment]
It works, thanks!
